components/Tabs/tests/unit/MDCTabBarLayoutTests.m

```diff
@@ -0,0 +1,136 @@
+}
+
+@interface MDCTabBarLayoutTests : XCTestCase
+
+@end
+
+@implementation MDCTabBarLayoutTests {
+  MDCTabBar *_tabBar;
+}
+
+- (void)setUp {
+  [super setUp];
+
+  // Create a tabbar with some dummy items and a dummy frame.
+  _tabBar = [[MDCTabBar alloc] initWithFrame:CGRectZero];
+  CGFloat tabBarHeight = [MDCTabBar defaultHeightForItemAppearance:_tabBar.itemAppearance];
+  _tabBar.frame = CGRectMake(0, 0, 200, tabBarHeight);
+  UITabBarItem *item1 = [[UITabBarItem alloc] initWithTitle:@"first" image:nil tag:0];
+  UITabBarItem *item2 = [[UITabBarItem alloc] initWithTitle:@"second" image:nil tag:0];
+  _tabBar.items = @[ item1, item2 ];
+  [_tabBar setNeedsLayout];
+  [_tabBar layoutIfNeeded];
+}
+
+- (void)tearDown {
+  // Explictly nil out ivars so that they are properly deallocated between tests. Otherwise XCTest
+  // will keep them alive until all test cases are complete.
+  _tabBar = nil;
+  [super tearDown];
+}
+
+// Tests the default layout behavior of of the tab bar's collection view.
+- (void)testLeftToRightLayout {
+  // Given
+  UICollectionView *collectionView = ExtractCollectionViewFromTabBar(_tabBar);
+  NSArray<UICollectionViewCell *> *sortedVisibleItems =
+      SortedCellsFromCollectionView(collectionView);
+  XCTAssertEqual(sortedVisibleItems.count, 2ul);
+  UICollectionViewCell *firstItemCell = sortedVisibleItems.firstObject;
+  UICollectionViewCell *secondItemCell = sortedVisibleItems.lastObject;
+  UICollectionViewFlowLayout *flowLayout =
+      (UICollectionViewFlowLayout *)collectionView.collectionViewLayout;
+
+  // When
+  _tabBar.mdf_semanticContentAttribute = UISemanticContentAttributeUnspecified;
+
+  // Then
+  CGFloat leftInset = flowLayout.sectionInset.left;
+  XCTAssertEqualWithAccuracy(firstItemCell.frame.origin.x, leftInset, (CGFloat)0.001);
+  XCTAssertEqualWithAccuracy(secondItemCell.frame.origin.x,
+                             leftInset + firstItemCell.frame.size.width, (CGFloat)0.001);
+}
+
+// Tests that setting UISemanticContentAttributeForceRightToLeft causes the tab bar's collection
+// view to lay out the cells in RTL (right-to-left) manner.
+- (void)testRightToLeftLayout {
+  // Given
+  UICollectionView *collectionView = ExtractCollectionViewFromTabBar(_tabBar);
+  UICollectionViewFlowLayout *flowLayout =
+      (UICollectionViewFlowLayout *)collectionView.collectionViewLayout;
+
+  // When
+  _tabBar.mdf_semanticContentAttribute = UISemanticContentAttributeForceRightToLeft;
+  [_tabBar setNeedsLayout];
+  [_tabBar layoutIfNeeded];
+
+  // Then
+  XCTAssertEqual(_tabBar.mdf_semanticContentAttribute, UISemanticContentAttributeForceRightToLeft);
+  NSArray<UICollectionViewCell *> *sortedVisibleItems =
+      SortedCellsFromCollectionView(collectionView);
+  XCTAssertEqual(sortedVisibleItems.count, 2ul);
+  UICollectionViewCell *firstItemCell = sortedVisibleItems.firstObject;
+  UICollectionViewCell *secondItemCell = sortedVisibleItems.lastObject;
+  CGFloat totalWidth = _tabBar.frame.size.width;
```

Comment:
Same comments here - please use `CGRectGet...` functions to retrieve the properties of the views' frames.

Reply:
Done.